[15.0][IMP] queue_job: Configure default subchannel capacity.

amh-mw opened the pull request:
This adds a new `subcapacity` option to channels that allows the
configuration of the default capacity for autocreated child channels.

For example, environment `ODOO_QUEUE_JOB_CHANNELS=root:8:subcapacity=1`
would set the capacity of an autocreated `root.sub` channel to 1.

amh-mw commented:
My own use case is that I run multi-company odoo with an environment like ODOO_QUEUE_JOB_CHANNELS=root:8,single:1. The single channel is for problem children that tend to generate an excess of SerializationFailure if executed in parallel. I've recently determined that those SerializationFailure are company-specific, so a channel per company would improve parallelism.

Per @sbidoul in OCA Discord
> Undeclared channels are auto created at runtime, but with an unlimited capacity.

This setting allows me to limit the capacity of auto-created channels.

sbidoul commented:
Thanks for addressing this @amh-mw 

I would have a preference for configuring this in the parent channel, instead of globally.
Like this is done, for instance for the `throttle` config.

amh-mw commented:
> I would have a preference for configuring this in the parent channel, instead of globally. Like this is done, for instance for the `throttle` config.

Understood. I will unfortunately not have time to circle back around to that request until next year. Would it be possible to mark this issue`nostale`?

sbidoul commented:
@amh-mw Count on me to repoen it if the stale bot closes it. ~Can you set it Draft?~ [done]

amh-mw committed:
[IMP] queue_job: Configure default subchannel capacity.

This adds a new `subcapacity` option to channels that allows the
configuration of the default capacity for autocreated child channels.

For example, environment `ODOO_QUEUE_JOB_CHANNELS=root:8:subcapacity=1`
would set the capacity of an autocreated `root.sub` channel to 1.

amh-mw commented:
@sbidoul Version two ready for review.

amh-mw commented:
Reopening against 18.0 https://github.com/OCA/queue/pull/767
